daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -482,6 +474,15 @@ class VariableMap private(vTable: Map[GlobalQName, ArrayBuffer[VariableInstance]
     Assert.invariant(variableInstances.isDefined)
 
     val v = vrd.createVariableInstance()
+
+    // If the variable is external, the original variable instance should have
+    // already been set to the externally provided value. Use this value as the
+    // default value
+    if (vrd.external && !vrd.hasLocalDefault) {
+      val value = variableInstances.get.last.externalValue
+      v.externalValue = value
+      v.setValue(value)
+    }
```

Comment:
This sort of splits the NVI logic into two separate places, which I think makes it a bit harder to reason about. Part is here, and part is the actual parser/unparse? Thoughts on moving this logic into to NVI parser/unparse so it's all consolidated in one place?

Also, thoughts on changing the ``externalValue`` to be ``maybeExternalValue``, and anytime we create a new variable instance we just copy that value from the previous instance to the new instance (you may already do this). This way, we always have the external value if we need it, and you never need to look at previous onces. So the NVI logic just becomes something like

```scala
if (needToEvaluateDefaultExpression) {
  // create suspension to set default value
} else if (v.maybeExternalValue.isDefined) {
  v.setValue(v.maybeExerternalValue.get)
} else {
  // no default anywhere and no external value, new instance remains undefined
}
```

This way, if logic about NVI's ever changes, we only have to change the one parser/unparser and don't also have to think about what other places might deal with external values/defaultExpressions/state/etc

Also note that this logic doesn't care if a variable is external or not. Only setExtVariable should care about that. Everything else can just assume if it has an external variable then it was allowed to be set.